Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/CopyPasteHelper.cs

```diff
@@ -0,0 +1,429 @@
+            {
+                indexOffset += 1;
+                foreach (var index in indicesToInsert)
+                    PasteBlocks(transmission, index + indexOffset, arrayToInsertInto);
+            }
+        }
+
+        private static void PasteBlocks(string transmission, int indexToInsert, SerializedProperty arrayToInsertInto)
+        {
+            var block = transmission.Substring(transmission.IndexOf(k_StartOfText, StringComparison.Ordinal) + 1);
+            var copiedType = GetCopiedClipboardType();
+            if (copiedType == typeof(InputActionMap))
+                PasteElement(arrayToInsertInto, block, indexToInsert, out _);
+            else if (copiedType == typeof(InputAction))
+                PasteAction(arrayToInsertInto, block, indexToInsert);
+            else
+            {
+                var actionName = Selectors.GetSelectedBinding(s_State)?.wrappedProperty.FindPropertyRelative("m_Action")
+                    .stringValue;
+                if (s_State.selectionType == SelectionType.Action)
+                    actionName = PropertyName(Selectors.GetSelectedAction(s_State)?.wrappedProperty);
+                PasteBindingOrComposite(arrayToInsertInto, block, indexToInsert, actionName);
+            }
+        }
+
+        private static SerializedProperty PasteElement(SerializedProperty arrayProperty, string json, int index, out string oldId, string name = "newElement",  bool changeName = true, bool assignUniqueIDs = true)
+        {
+            var duplicatedProperty = AddElement(arrayProperty, name, index);
+            duplicatedProperty.RestoreFromJson(json);
+            oldId = duplicatedProperty.FindPropertyRelative("m_Id").stringValue;
+            if (changeName)
+                EnsureUniqueName(duplicatedProperty);
+            if (assignUniqueIDs)
+                AssignUniqueIDs(duplicatedProperty);
+            s_lastAddedElement = duplicatedProperty;
+            return duplicatedProperty;
+        }
+
+        private static void PasteAction(SerializedProperty arrayProperty, string jsonToInsert, int indexToInsert)
+        {
+            var json = jsonToInsert.Split(k_BindingData, StringSplitOptions.RemoveEmptyEntries);
+            var bindingJsons = json[1].Split(k_EndOfBinding, StringSplitOptions.RemoveEmptyEntries);
+            var property = PasteElement(arrayProperty, json[0], indexToInsert, out _, "");
+            var newName = PropertyName(property);
+            var newId = property.FindPropertyRelative("m_Id").stringValue;
+            var actionMapTo = Selectors.GetActionMapForAction(s_State, newId);
+            var bindingArrayToInsertTo = actionMapTo.FindPropertyRelative(nameof(InputActionMap.m_Bindings));
+            var prevActionName = PropertyName(arrayProperty.GetArrayElementAtIndex(indexToInsert - 1));
+            var index = bindingArrayToInsertTo.LastOneThatFits(b => b.FindPropertyRelative("m_Action").stringValue.Equals(prevActionName)).GetIndexOfArrayElement() + 1;
```

Comment:
Is this really safe? According to code above it seems LastOneThatFits may return null?

Reply:
good catch - adjusted to find a index iterative